measure

```diff
@@ -9,86 +9,112 @@ import yaml
+            driver_key = os.path.basename(__file__)
         try:
-            config = yaml.safe_load(open(path))['prom']
-            assert config and config.get('metrics'), 'No metrics has been provided at path "prom.metrics" ' \
-                                                     'in file located at {}'.format(path)
+            config = yaml.safe_load(open(path))[driver_key]
+            assert config and config.get("metrics"), 'No metrics has been provided at path "{}.metrics" in file located at {}"'.format(driver_key, path)
         except yaml.YAMLError as e:
-            e.__cause__ = 'Error parsing config file located at {}: {}'.format(path, e.__cause__)
+            e.__cause__ = "Error parsing config file located at {}: {}".format(
+                path, e.__cause__
+            )
             raise
         except KeyError:
-            raise KeyError('No `prom` configuration has been provided in config file located '
-                           'at {}'.format(path))
+            raise KeyError("No `{}` configuration has been provided in config file located at {}".format(driver_key, path))
 
         # validate config
-        bad_top_keys = config.keys() - {'metrics'}
+        bad_top_keys = config.keys() - {"metrics", "prometheus_endpoint"}
         if bad_top_keys:
-            raise Exception('Unknown top level key(s) in `prom` section of file {}: {}'.format(path, bad_top_keys))
-        metrics = config['metrics']
-        assert isinstance(metrics, dict), 'The `metrics` attribute in `prom` section must be a dict'
-        for m, mdesc in config['metrics'].items():
-            assert isinstance(m, str), 'Metric name `{}` must be a string, found {}'.format(m, type(m).__name__)
+            raise Exception("Unknown top level key(s) in `{}` section of file {}: {}".format(driver_key, path, bad_top_keys))
+
+        # Validate prometheus_endpoint, if present
+        if "prometheus_endpoint" in config:
+            if not isinstance(config["prometheus_endpoint"], str):
+                raise Exception("'prometheus_endpoint' in '{}' section must be a string".format(driver_key))
+            if not config["prometheus_endpoint"]:
+                raise Exception("'prometheus_endpoint' in '{}' section must not be empty".format(driver_key))
+
+        metrics = config["metrics"]
+        assert isinstance(metrics, dict), "The `metrics` attribute in `{}` section must be a dict".format(driver_key)
+        for m, mdesc in config["metrics"].items():
+            assert isinstance(m, str), "Metric name `{}` must be a string, found {}".format(m, type(m).__name__)
             if not isinstance(mdesc, dict):
-                raise Exception('Metric `{}` data in file {} must be a dict, found {}'.
-                    format(m, path, type(mdesc).__name__))
-            bad_keys = mdesc.keys() - {'query', 'unit'}
+                raise Exception(
+                    "Metric `{}` data in file {} must be a dict, found {}".format(
```

Comment:
I would name `data` - `settings`

Reply:
Can you please explain what do you mean ?